wormhole_chain/x/wormhole/keeper/keeper.go

```diff
@@ -37,6 +40,17 @@ func NewKeeper(
 	}
 }
 
+// This is necessary because x/staking relies on x/wormhole and x/wasmd relies on x/staking,
+// So we must either:
+// 1. make wormhole depend on staking and replace the modified functions from here.
+// 2. add a new module that wraps x/wasmd instead of using x/wormhole.
+// 3. (current) set wasmdKeeper late in init and use guards whenever it's referenced.
+// Opted for (3) as we only reference in two places.
+func (k *Keeper) SetWasmdKeeper(keeper wasmdkeeper.Keeper) {
+	k.wasmdKeeper = keeper
```

Comment:
I don't think it's an issue if the keeper is set twice.  The Keeper objects can be copied around freely with little issue, but it could be probably if one isn't fully initialized, which is the risk with something like `SetWasmdKeeper`.